packages/sw-cli/src/lib/generate-file-manifest.js

```diff
@@ -0,0 +1,53 @@
+const getFileManifestEntries = require('./get-file-manifest-entries');
+const writeFileManifest = require('./utils/write-file-manifest');
+const errors = require('./errors');
+
+/**
+ * @example <caption>Generate a service worker for a project.</caption>
+ * const swCLI = require('sw-cli');
+ *
+ * swCLI.generateFileManifest({
+ *   manifestFilePath: './build/manifest.js'
+ *   rootDirectory: './build/',
+ *   globPatterns: ['**\/*.{html,js,css}'],
+ *   globIgnores: ['admin.html']
+ * })
+ * .then(() => {
+ *   console.log('Build Manifest generated.');
+ * });
+ *
+ * This method will generate a file manifest that can be used in a service
+ * worker for caching assets offline..
+ * @param {Object} input
+ * @param {String} input.manifestFilePath The name you wish to give to your
+ * manifest file.
+ * @param {String} input.rootDirectory The root of the files you wish to
+ * be cached. This will also be the directory the service worker and library
+ * files are written to.
+ * @param {Array<String>} input.globPatterns Patterns to glob for when
+ * generating the build manifest.
+ * @param {String|Array<String>} input.globIgnores Patterns to exclude when
+ * generating the build manifest.
+ * @return {Promise} Resolves once the service worker has been generated
+ * with a precache list.
+ *
+ * @memberof module:sw-cli
+ */
+const generateFileManifest = (input) => {
+  if (!input || typeof input !== 'object' || input instanceof Array) {
+    return Promise.reject(
+      new Error(errors['invalid-generate-file-manifest-arg']));
+  }
+
+  const manifestFilePath = input.manifestFilePath;
```

Comment:
I'm not sold on `manifestFilePath` as the parameter name here. I feel like something with `output` or `dest` would be more explicit.

(I'm looking at what `webpack` currently does, since folks might be familiar with that, and they [use](https://webpack.js.org/guides/get-started/#using-webpack-with-a-config) `output: {filename, path}`, though breaking it up into separate properties for the file and path is overkill for us.)